azext_iot/assets/user_messages.py

```diff
@@ -10,13 +10,13 @@
 }
 
 
-def error_no_hub_or_login_on_input(entity_type="IoT Hub"):
-    if entity_type:
+def error_no_hub_or_login_on_input(entity_type: str = "IoT Hub"):
+    if entity_type and isinstance(entity_type, str):
```

Comment:
what is the reason for checking if the instance is an str? When should this be called with something that isnt an str? (the type hint in the param should tell users that entity_type is str so it should not be an issue)

Reply:
To reduce mistakes and providing a clear dev error. The type hint, isinstance, and TypeError were refactored-in, in prior iteration of this PR. We aren't enforcing type checks/constraints in CI, if you think there is no chance of an issue (like there was here in the past), then I'm fine removing the isinstance check.

Reply:
Talked with @vilit1 offline, we'll keep it.